### ISSUE TITLE
Allow inserting text after code snippets

### ISSUE THREAD

**Issue by fahhem** [Reviewable/Reviewable#1095]
Code snippets and suggestions currently have to be the last item(s) in a comment, but it's often desirable to have a snippets interspersed with some text, such as in the following comment:

> ```py
> data = []
> for i in range(3):
>   x = lst[i]
>   data.append(f(x))
> ```
> Let's improve this by iterating over `lst` correctly:
> ```py
> data = []
> for x in lst:
>   data.append(f(x))
> ```
> And we can improve it further by just using `map()`:
> ```py
> data = map(f, lst)
> ```
> Or more readable-y as a list comprehension:
> ```py
> data = [f(x) for x in lst]
> ```

Which flows a lot better than what's currently allowed:

> ```py
> data = []
> for i in range(3):
>   x = lst[i]
>   data.append(f(x))
> ```
> Let's improve this by iterating over `lst` correctly. And we can improve it further by just using `map()`. Or more readable-y as a list comprehension.
> ```py
> data = []
> for x in lst:
>   data.append(f(x))
> ```
> ```py
> data = map(f, lst)
> ```
> ```py
> data = [f(x) for x in lst]
> ```

**Comment by jwnimmer-tri** [comment]
For my part, I like the simplicity of the current discussion interface.

If I want to do code quotes inline, I can (and do) just use inline markdown:
```
  Here's an idea:

  ```python
  foo = bar()
  ...```

  or instead

  ```python
  foo = baz()
  ...```
```

>   Here's an idea:
>
>  ```python
>  foo = bar()
>  ```
>
>  or instead
>
>  ```python
>  foo = baz()
>  ```

**Comment by pkaminski** [comment]
I agree with @jwnimmer-tri -- I'm not convinced the extra UI and implementation complexity is worth it, when you can already get a mixed flow using inline code quotes when it's really needed.  Also, the code blocks are automatically numbered so the "bad" example could be rewritten as the following, which I think reads just fine:
> Let's improve this by iterating over `lst` correctly as in (i). Then we can improve it further by (ii) just using `map()`, or make it even more readable as a (iii) list comprehension.
